devito/operator/operator.py

```diff
@@ -305,26 +305,9 @@ def _lower_exprs(cls, expressions, **kwargs):
             else:
                 dimension_map = {}
 
-            mapper = {}
-
             # Handle Functions (typical case)
-            for f in retrieve_functions(expr):
-                # Get spacing symbols for replacement
-                spacings = [i.spacing for i in f.dimensions]
-
-                # Only keep the ones used as indices
-                spacings = [s for i, s in enumerate(spacings)
-                            if s.free_symbols.intersection(f.args[i].free_symbols)]
-
-                # Substitution for each index
-                subs = {**{s: 1 for s in spacings}, **dimension_map}
-
-                # Introduce shifting to align with the computational domain,
-                # and apply substitutions
-                indices = [(a - i + o).xreplace(subs)
-                           for a, i, o in zip(f.args, f.origin, f._size_nodomain.left)]
-
-                mapper[f] = f.indexed[indices]
+            mapper = {f: f.indexify(w_shift=True, extra_subs=dimension_map)
```

Comment:
The reason this was "duplicated" is that the Operator version was optimized for speed, while `indexify` was not, but I couldn't just drop it for backwards compatibility